base/src/main/java/proguard/evaluation/stateTrackers/PartialEvaluatorStateTracker.java

```diff
@@ -0,0 +1,94 @@
+package proguard.evaluation.stateTrackers;
+
+import proguard.classfile.Clazz;
+import proguard.classfile.Method;
+import proguard.classfile.attribute.CodeAttribute;
+import proguard.classfile.attribute.ExceptionInfo;
+import proguard.classfile.instruction.Instruction;
+import proguard.evaluation.BasicBranchUnit;
+import proguard.evaluation.PartialEvaluator;
+import proguard.evaluation.TracedStack;
+import proguard.evaluation.TracedVariables;
+import proguard.evaluation.Variables;
+import proguard.evaluation.value.InstructionOffsetValue;
+
+import java.util.List;
+
+public interface PartialEvaluatorStateTracker
+{
+    /************************
+     * Code attribute level *
+     ************************/
```

Comment:
```suggestion
    /**
     * Code attribute level
     */
```

Can you use the standard formatting like this? 

Can you add to each method in the interface the Java doc with some short explanation.